### PR TITLE
add benchmark: copy, symlink, hardlink

### PR DESCRIPTION
**Pull request by pared**
Data is randomly generated.
Next step should include real use case data. Probably from our own dataset repo.

Will create issue after this one gets accepted.

### CODE REVIEW

File: benchmarks/add.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+class Add_100_1M_copy(BaseBench):
+    def setup(self):
+        super().setup()
+        self.repo = init_dvc(self.test_directory.name)
+        dataset_path = random_data_dir(100, 1024 * 1024)
+        shutil.copytree(dataset_path, "data")
+
+    def time_add(self):
+        main(["add", "data"])
```
Comment (efiop) [inline benchmarks/add.py:16]:
```suggestion
        assert main(["add", "data"]) == 0
```

File: benchmarks/base.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+def sources_dir():
+    path = os.path.join(gettempdir(), "sources")
+    os.makedirs(path, exist_ok=True)
+    return path
+
+
+class BaseBench:
+    def setup(self):
+        self.cwd = os.getcwd()
+        self.test_directory = TemporaryDirectory(prefix="HelpSuite")
```
Comment (efiop) [inline benchmarks/base.py:16]: Why `HelpSuite` though?

File: benchmarks/add.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+class Add_100_1M_copy(BaseBench):
+    def setup(self):
+        super().setup()
+        self.repo = init_dvc(self.test_directory.name)
+        dataset_path = random_data_dir(100, 1024 * 1024)
+        shutil.copytree(dataset_path, "data")
+
+    def time_add(self):
+        assert main(["add", "data"]) == 0
```
Comment (efiop) [inline benchmarks/add.py:16]: As we've discussed, `main` will launch stuff like updater/analytics workers, which might affect the results. We need to check that.

File: benchmarks/base.py (hunk 2 above)
Comment (pared) [reply benchmarks/base.py]: Leftover from initial version, sorry.

File: benchmarks/add.py (hunk 3 above)
Comment (pared) [reply benchmarks/add.py:16]: Turning analytics off during initialization.

**Pull request merged**